src/rotp/Rotp.java

```diff
@@ -49,9 +51,18 @@
     private static float resizeAmt =  -1.0f;
     public static int actualAlloc = -1;
     public static boolean reloadRecentSave = false;
+    private static GraphicsDevice device = GraphicsEnvironment.getLocalGraphicsEnvironment().getScreenDevices()[0];
     
     public static void main(String[] args) {
         frame = new JFrame("Remnants of the Precursors");
+        
+       if (UserPreferences.getPreferenceValueFromFile("FULL_SCREEN").equalsIgnoreCase("YES"))
```

Comment:
I don't like getting the full screen option directly from the settings file, specially here, but i had no other options (without changing more code) because at this point the preferences object is not initialized yet.

Reply:
How are you preventing stretching of the background artwork? That's the primary reason the game using a fixed aspect ratio

Reply:
The game is not stretched to fill the resolution (width and height) of the screen. On full screen mode, it simply uses the screenSizePct given by setScreenSizePct(100). On my native resolution (1680x1050), i have no issues with this and get no black borders, but on other resolutions you may get them.

Reply:
I really hope we will get full screen support one day